BLSQ/iaso
# WC2-361: Calculations for average weight gain

Explain what problem this PR is resolving

Related JIRA tickets : [WFPPM-16](https://bluesquare.atlassian.net/browse/WFPPM-16)

## Self proofreading checklist


## Changes
Added new fields in the Journey table:
   

- initial weight: the weigth at the starting of the journey
-    discharge weight: the weigth on exiting the program
-    start date: the admission date
-    end date: the exit date
-    duration: number of days between admission and exiting
-    weigth loss: the difference between discharge weight and initial weight weight when it's negative    
-    weight_gain: the average of weight gain when the the difference between discharge weight and initial weight weight when is positive


## How to test
From the local instance with wfp coda database, Lunch with docker-compose run iaso manage wfp_etl the ETL script in order to generate data for tableau dashboard and go to the debug page for the beneficiary: e.g: http://localhost:8081/wfp/debug/325/
You should see the calulcated average weight gain. 

e.g: http://localhost:8081/wfp/debug/325/   
          - initial weight = 8 Kgs
          - discharge weight = 10 Kgs
          - start date = 02 Feb 2024
          - end date = 1 March 2024
          - duration = days between start and end date = 28 days
          - weight_gain = (discharge weight - initial weight) / (initial weight x initial weight) ==> (10 - 8) / (8 x 28) = 0.0089 Kgs
          
![Sélectionnez-l’objet-journey-à-changer-Iaso](https://github.com/BLSQ/iaso/assets/6383219/d431c120-9ebb-4295-9e7d-7683b88ce646)
 



## Print screen / video
   - Added fields in Journey table
![image](https://github.com/BLSQ/iaso/assets/6383219/ff900a2a-3394-482d-8959-45cb6be421f4)

  - Showing new fields in debug page: e.g: http://localhost:8081/wfp/debug/325/
       
   ![showing-additional-fields-wfp-debug-325](https://github.com/BLSQ/iaso/assets/6383219/eae0e753-bc78-4201-8393-29180dc56c64)




[WFPPM-16]: https://bluesquare.atlassian.net/browse/WFPPM-16?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
